File: src/main/java/sk/matusko/wixit/authentication/services/UserService.java
package sk.matusko.wixit.authentication.services;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.validation.annotation.Validated;
import sk.matusko.wixit.authentication.dao.User;

import javax.validation.Valid;

@Validated
public interface UserService {

    User save(@Valid User user);

    Page<User> all(Pageable pageable);

    Page<User> filterByLastName(String lastName, Pageable pageable);

}

File: src/main/kotlin/sk/matusko/wixit/auth/service/UserService.kt
package sk.matusko.wixit.auth.service

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.validation.annotation.Validated
import sk.matusko.wixit.common.dao.User
import javax.validation.Valid

@Validated
interface UserService {

    fun save(@Valid user: User): User

    fun all(pageable: Pageable): Page<User>

    fun filterByLastName(lastName: String, pageable: Pageable): Page<User>

}